arm9/overlays/24/src/ov24_02254840.c

```diff
@@ -0,0 +1,254 @@
+    if (appHandler->unk2 == 0) {
+        appHandler->unk0 = arg1;
+    }
+    else {
+        appHandler->unk0 = 2;
+    }
+    appHandler->unk1 = 0;
+}
+
+BOOL ov24_022549AC(MemoPadAppHandler* appHandler) {
+    switch (appHandler->unk1) {
+        case 0:
+            ov24_02254D8C(appHandler->displayHandler, 0);
+            appHandler->unk1++;
+            break;
+        case 1:
+            if (ov24_02254DB0(appHandler->displayHandler, 0)) {
+                ov20_0225298C(appHandler->unk16F8);
+                ov24_02254998(appHandler, 1);
+            }
+            break;
+        default:
+            break;
+    }
+    return FALSE;
+}
+
+BOOL ov24_022549F8(MemoPadAppHandler* appHandler) {
+    if (appHandler->unk2 != 0) {
+        ov24_02254998(appHandler, 2);
+    }
+    switch (appHandler->unk1) {
+    case 0:
+        if (ov20_02252C08(appHandler->unk16F8)) {
+            break;
+        }
+        if (appHandler->unk3) {
+            u32 x = appHandler->drawState.lastModifiedX;
+            u32 y = appHandler->drawState.lastModifiedY;
+            if (ov24_02254AD4(appHandler)) {
+                ov24_02254B20(appHandler, x, y, appHandler->drawState.lastModifiedX, appHandler->drawState.lastModifiedY);
+            } else {
+                appHandler->unk3 = 0;
+            }
+        } else {
+            if (ov24_02254A70(appHandler)) {
+                ov24_02254D8C(appHandler->displayHandler, 3);
+                appHandler->unk3 = 1;
+            }
+        }
+        break;
+    }
+    return FALSE;
+}
+
+BOOL ov24_02254A70(MemoPadAppHandler* appHandler) {
+    u32 x, y;
+    if (TouchScreen_GetTapState(&x, &y)) {
+        if (((x - 16) < 156) & ((y - 16) < 150)) {
+            x = (x - 16) >> 1;
+            y = (y - 16) >> 1;
+            if (appHandler->drawState.pixelData[x][y] != appHandler->drawState.stylusType) {
+                appHandler->drawState.pixelData[x][y] = appHandler->drawState.stylusType;
+                appHandler->drawState.lastModifiedX = x;
+                appHandler->drawState.lastModifiedY = y;
+                return TRUE;
+            }
+         }
+    }
+    return FALSE;
+}
+
+BOOL ov24_02254AD4(MemoPadAppHandler* appHandler) {
+    u32 x, y;
+    if (TouchScreen_GetTapState(&x, &y)) {
+        if (((x - 16) < 156) & ((y - 16) < 150)) {
+            x = (x - 16) >> 1;
+            y = (y - 16) >> 1;
+            appHandler->drawState.lastModifiedX = x;
+            appHandler->drawState.lastModifiedY = y;
+            return TRUE;
+        }
+    }
+    return FALSE;
+}
+
+void ov24_02254B20(MemoPadAppHandler* appHandler, u32 x0, u32 y0, u32 x1, u32 y1) {
+    if (y0 == y1 && x0 == x1) {
+        return;
+    }
+    u32 dx = x1 > x0 ? x1 - x0 : x0 - x1;
+    u32 dy = y1 > y0 ? y1 - y0 : y0 - y1;
+    s32 direction, offset;
+    fx32 f, g;
+    if (dx > dy) {
+        direction = (x1 > x0) ? 1 : -1;
+        if (dx != 0) {
+            f = ((y1 - y0) * 0x1000);
+            f /= (s32)dx;
```

Comment:
gf moment...

Reply:
I would wager there's a better way to write most of this function but I'm entirely unfamiliar with fx32 so this is just what matches

Reply:
yeah fair enough, fixed points are not that fun